confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import logging
+
+from .schema import SchemaProvider
+
+log = logging.getLogger(__name__)
+
+"""
+API Endpoint templates
+"""
+SCHEMAS = '{}/schemas/ids/{}?format=serialized'
+SUBJECT = '{}/subjects/{}'
+SUBJECTS = '{}/subjects'
+VERSION = '{}/subjects/{}/versions/{}'
+VERSIONS = '{}/subjects/{}/versions'
+CONFIG = '{}/config/{}'
+CONFIGS = '{}/config'
+
+
+class SchemaRegistryClient(object):
+    """
+    A client that talks to a Schema Registry over HTTP
+
+    See Also:
+        http://confluent.io/docs/current/schema-registry/docs/intro.html
+
+    Arguments:
+        conf (SchemaRegistryConfig): Schema Registry config object
+    """
+
+    def __init__(self, conf):
+        self.rest_client = conf.RestClient
+        self.id_to_schema = {}
+
+    def __del__(self):
+        if self.rest_client is not None:
+            self.rest_client.close()
```

Comment:
one reason rest client should be completely encapsulated is that then it's obvious where ownership lies. by calling `close` here you're specifying that you are transferring ownership to SchemaRegistryClient, which is ok... if document, but this could be a non-issue.

Reply:
I missed this, good point